source/nodejs/adaptivecards-templating/src/template-engine.ts

```diff
@@ -240,17 +228,25 @@ export class Template {
             }
         }
 
-        // If there are none, it's just a string
-        if (matchCount === 0) {
-            return interpolatedString;
-        }
-        // If the entire string is enclosed in a single ${}, extract the enclosed expression
-        else if (matchCount === 1 && lastMatch[0].length === interpolatedString.length) {
-            return AEL.Expression.parse(lastMatch[1], lookup);
+        // If there is at least one expression start marker, let's attempt to convert into an expression
+        if (interpolatedString.indexOf("${") >= 0) {
+            let parsedExpression = AEL.Expression.parse("`" + interpolatedString + "`", lookup);
+
+            if (parsedExpression.type === "concat") {
+                if (parsedExpression.children.length === 1 && !(parsedExpression.children[0] instanceof AEL.Constant)) {
+                    // The concat contains a single child that isn't a constant, thus the original
+                    // string was a single expression. When evaluated, we want it to produce the type
+                    // of that single expression
+                    return parsedExpression.children[0];
+                }
+
+                // Otherwise, we want the expression to produce a string
+                return parsedExpression;
+            }
```

Comment:
If `parsedExpression.type` is not `concat`, do we want to return the unparsed string?

Reply:
That's what the code does.